src/dotnet/SlnFileExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli.Sln.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    public static class SlnFileExtensions
+    {
+        public static IList<string> GetSolutionFoldersFromProject(this SlnFile slnFile, SlnProject project)
```

Comment:
Ok thanks. I'll create two new files SlnProjectExtensions and SlnProjectCollectionExtensions.